stellar-contract-env-common/src/env.rs

```diff
@@ -175,26 +175,35 @@ macro_rules! call_macro_with_all_host_functions {
                 {"0", fn bigint_to_u64(x:Object) -> u64}
                 {"1", fn bigint_from_i64(x:i64) -> Object}
                 {"2", fn bigint_to_i64(x:Object) -> i64}
-                {"3", fn bigint_add(x:Object, y:Object) -> Object}
-                {"4", fn bigint_sub(x:Object, y:Object) -> Object}
-                {"5", fn bigint_mul(x:Object, y:Object) -> Object}
-                {"6", fn bigint_div(x:Object, y:Object) -> Object}
-                {"7", fn bigint_rem(x:Object, y:Object) -> Object}
-                {"8", fn bigint_and(x:Object, y:Object) -> Object}
-                {"9", fn bigint_or(x:Object, y:Object) -> Object}
-                {"A", fn bigint_xor(x:Object, y:Object) -> Object}
-                {"B", fn bigint_shl(x:Object, y:i64) -> Object}
-                {"C", fn bigint_shr(x:Object, y:i64) -> Object}
-                {"D", fn bigint_cmp(x:Object, y:Object) -> RawVal}
-                {"E", fn bigint_is_zero(x:Object) -> RawVal}
-                {"F", fn bigint_neg(x:Object) -> Object}
-                {"G", fn bigint_not(x:Object) -> Object}
-                {"H", fn bigint_gcd(x:Object, y:Object) -> Object}
-                {"I", fn bigint_lcm(x:Object, y:Object) -> Object}
-                {"J", fn bigint_pow(x:Object, y:RawVal) -> Object}
-                {"K", fn bigint_pow_mod(p:Object, q:Object, m:Object) -> Object}
-                {"L", fn bigint_sqrt(x:Object) -> Object}
-                {"M", fn bigint_bits(x:Object) -> u64}
+                /// Creates a BigInt from a binary array `x` and i32 sign `s`.
+                /// Bytes in big-endian order. Sign is interpreted: -1 as negative, 0 as zero, 1 as positive
+                /// If sign is 0, then the input bytes are ignored and will return a BigInt of 0.
+                {"3", fn bigint_from_bytes_be(x:Object, s:RawVal) -> Object}
```

Comment:
Do we really need these functions? They aren't mentioned in CAP-0051, and I'm not entirely sure how they would be used right now.

Part of my concern is that these interfaces seem a bit like foot-guns. For example, should we ignore `x` if `s == 0` or should we trap? I can see that the library handles the case where `x` is empty but `s != 0` by making the result 0, but should we trap in that case? It's all pretty unclear to me.

Also based on other conventions you've been using I would expect `s` to be `u32`.

In summary, there are a lot of questions here. I think you should drop this commit for now and we should work out these details in the CAP first, instead of trying to work them out in code where it is much harder and more time consuming.